### DISCUSSION TITLE
GetMany?

### DISCUSSION THREAD

**Discussion by famousfilm** [deta/cloud-docs#565] [Ideas] [answered]
Service looks cool and I'd like to try it out. How do you perform a `bulk get` or `get many`? I see `get` [based on key in the SDK docs](https://docs.deta.sh/docs/base/sdk/#get) and see a `put many` as well as a `delete many` [elsewhere in the docs](https://docs.deta.sh/docs/drive/sdk#delete-many). Does `get many` exist? It seems like looping over _n_ keys to get the data would be... bad. 😁

**Comment by jnsougata** [comment]
as you have n keys you can just do async request to base.... you have to implement with base's [http api](https://docs.deta.sh/docs/base/http)

**Comment by AyushSehrawat** [comment, marked as answer]
Use `fetch`

https://docs.deta.sh/docs/base/sdk#fetch

**Reply by jnsougata** [reply]
lol

**Reply by AyushSehrawat** [reply]
Lol ig you misinterpreted the question or your brain is working at a speed higher than light so it went backwards 😐

**Reply by jnsougata** [reply]
read the question properly nub

**Reply by AyushSehrawat** [reply]
"Does get many exist` - Yes use fetch

Looping over n keys? Just use fetch...

**Reply by jnsougata** [reply]
I see that's the issue with dumb people when they don't even know the context and reply
read it https://docs.deta.sh/docs/base/queries/#querying-keys and stfu

**Reply by AyushSehrawat** [reply]
Ok

**Reply by famousfilm** [reply]
`getMany == fetch`. Got it. 🤦 RTFM. Appreciate your responses. 😆

**Reply by jnsougata** [reply]
> `getMany == fetch`. Got it. facepalm RTFM. Appreciate your responses. laughing

well let's see how you can `fetch` `n` keys at once by using `fetch` good luck

**Reply by famousfilm** [reply]
Thanks for the doc references. That's true @jnsougata, if I'm querying for `n` keys, it looks like I _need_ to use querying keys. Practically speaking, I could use fetch for a single query if the keys start with the same prefix, which I can make work.

A better way to ask what my original "get many" question is, "How do I get an `IN` operator (vs having to use _n_ `OR` operators)?"

An example from the `OR` operator in the [querying keys docs](https://docs.deta.sh/docs/base/queries/#or) was that these two are equal:

```
[{"age?lte": 30}, {"age?gte": 40}]
SELECT * FROM base WHERE age<=30 OR age>=40;
```

If i'm looking up 30 keys, I don't want to use an `OR` clause between 30 keys, I want an `IN` clause, but I don't see that as an option.

Both of you have a point. I think what @jnsougata is saying is, "If you _actually_ want to do it by key (like you asked), you gotta use a bunch of `OR` statements." I think @AyushSehrawat is saying is "you probably want `fetch` though."

**Reply by jnsougata** [reply]
> Thanks for the doc references. That's true @jnsougata, if I'm querying for `n` keys, it looks like I _need_ to use querying keys. Practically speaking, I could use fetch for a single query if the keys start with the same prefix, which I can make work.
> 
> A better way to ask what my original "get many" question is, "How do I get an `IN` operator (vs having to use _n_ `OR` operators)?"
> 
> An example from the `OR` operator in the [querying keys docs](https://docs.deta.sh/docs/base/queries/#or) was that these two are equal:
> 
> ```
> [{"age?lte": 30}, {"age?gte": 40}]
> SELECT * FROM base WHERE age<=30 OR age>=40;
> ```
> 
> If i'm looking up 30 keys, I don't want to use an `OR` clause between 30 keys, I want an `IN` clause, but I don't see that as an option.
> 
> Both of you have a point. I think what @jnsougata is saying is, "If you _actually_ want to do it by key (like you asked), you gotta use a bunch of `OR` statements." I think @AyushSehrawat is saying is "you probably want `fetch` though."

`fetch` is never an option if your keys don't share same prefix. if you have n diff keys you have to go with async approach

**Reply by AyushSehrawat** [reply]
> > Thanks for the doc references. That's true @jnsougata, if I'm querying for `n` keys, it looks like I _need_ to use querying keys. Practically speaking, I could use fetch for a single query if the keys start with the same prefix, which I can make work.
> > 
> > A better way to ask what my original "get many" question is, "How do I get an `IN` operator (vs having to use _n_ `OR` operators)?"
> > 
> > An example from the `OR` operator in the [querying keys docs](https://docs.deta.sh/docs/base/queries/#or) was that these two are equal:
> > 
> > ```
> > [{"age?lte": 30}, {"age?gte": 40}]
> > SELECT * FROM base WHERE age<=30 OR age>=40;
> > ```
> > 
> > If i'm looking up 30 keys, I don't want to use an `OR` clause between 30 keys, I want an `IN` clause, but I don't see that as an option.
> > 
> > Both of you have a point. I think what @jnsougata is saying is, "If you _actually_ want to do it by key (like you asked), you gotta use a bunch of `OR` statements." I think @AyushSehrawat is saying is "you probably want `fetch` though."
> 
> `fetch` is never an option if your keys don't share same prefix. if you have n diff keys you have to go with async approach

Try sougata's way then!

**Reply by jnsougata** [reply]
@famousfilm  if you are using python you can try (https://github.com/jnsougata/deta) it... as I've implemented `get_multiple` there and it's fast enough to meet your requirements
![image](https://user-images.githubusercontent.com/53375272/199300496-3760f06d-ffd8-41d3-9fe3-023947191c17.png)
